### DISCUSSION TITLE
Work Model

### DISCUSSION THREAD

**Discussion by MontyPud** [macxq/foxess-ha#195] [Ideas]
Hello Eric.  I noticed on my FoxCloud mobile app on the real time data view there is a Work Model.  I think it is new or maybe I've never noticed it before.

It allows me to change the energy priorities.  It has so far been in self use mode, which is:
1. my home load
2. charge the battery
3. surplus to the grid

We now have Feed-in priority and Backup.  
Feed-in priority allows me to change the priorities to:
1. my home load
2. to the grid
3. surplus (like when the grid is full? :-) ) to the battery.

Backup is for islanded systems.

This is great because my tariff pays me more for exported electricity between 1600 and 1900.  I also get more money for feeding in than it costs me to charge my batteries if I do it between 0200 and 0500.

So, on a sunny day, I'd like my HA to do some sums and figure out the best time to start feeding to the grid rather than changing the battery.  I really don't want to run out of battery though, so through the winter I could see this setting being changed often, which would be a faff.

Please could you see if HA could be made to adjust this setting?  On the app and through the Fox ESS website, it wants me to enter my password, so that might make it awkward?

Thanks for all the work you've done so far Eric.  Really appreciate it.

Alec

Please could

**Comment by Grzble** [comment]
Hi. Any news here?  is it currently possible to control the work model?. I would like to create automation of the inverter working model with the RCE energy price state. thank you in advance for the information.

**Reply by Daibutt** [reply]
I'm not sure this cloud integration allows writing back to the inverter. 
As you presumably have HA up and running have you thought about Modbus RS485 connection to your inverter?
Fairly simple wiring addition and purchase of a relatively cheap RS485/Ethernet adapter will get you real time data and the ability to change work mode and various other inverter parameters.
https://github.com/nathanmarlor/foxess_modbus
The Wiki for this integration has a load of useful info and detailed setup instructions.
